cmd/garm/main.go

```diff
@@ -166,6 +168,28 @@ func main() {
 		}
 	}()
 
+	if !cfg.APIServer.MetricsConfig.Disabled {
+		go func() {
+
+			metricsMiddleware := auth.NewMetricsMiddleware(cfg.JWTAuth)
+
+			r := mux.NewRouter()
+			r.Handle("/metrics", promhttp.Handler())
+			if !cfg.APIServer.MetricsConfig.NoAuth {
+				r.Use(metricsMiddleware.Middleware)
+			}
+
+			err := prometheus.Register(controllers.NewGarmCollector(controller))
+			if err != nil {
+				log.Printf("failed to register prometheus collector: %+v", err)
+			}
+
+			if err := http.ListenAndServe(cfg.APIServer.MetricsBindAddress(), r); err != nil {
```

Comment:
Please use the existing listener. If a separate port is desired for metrics, nginx can be used as a reverse proxy to tweak how garm is exposed.